revng-test-configure

```diff
@@ -76,22 +76,23 @@ def most_similar_tuples(inputs, command_type):
             if index == longest_input:
                 continue
 
-            # A list of (has_different_source, -common_tags_count, index)
+            # A list of (has_different_source, -common_tags_count, different_tag_count, index)
             scores = []
             for candidate_index, candidate in enumerate(inputs[index]):
                 scores.append((reference.source_path != candidate.source_path,
                                -len(set(reference.tags) & set(candidate.tags)),
+                               len(set(reference.tags) ^ set(candidate.tags)),
                                candidate_index))
             scores = sorted(scores)
 
             if len(scores) > 1 and scores[0][:-1] == scores[1][:-1]:
-                error(f"While emitting {command_type}, {reference.path} has multiple good matches for index #{index}:\n  {inputs[index][scores[0][2]].path}\n  {inputs[index][scores[1][2]].path}")
+                error(f"While emitting {command_type}, {reference.path} has multiple good matches for index #{index}:\n  {inputs[index][scores[0][3]].path}\n  {inputs[index][scores[1][3]].path}")
 
             if not scores:
                 error(f"While emitting {command_type}, cannot match {reference.path} with any other target for input #{index}")
 
-            used_inputs.add((index, scores[0][2]))
-            entry[index] = inputs[index][scores[0][2]]
+            used_inputs.add((index, scores[0][3]))
+            entry[index] = inputs[index][scores[0][3]]
```

Comment:
Please thoroughly elaborate. I'm very sensitive to changes in this area.

Reply:
Ah, I was planning to do a more explicit commit message for this but I guess I just forgot.

The gist is, if there is a situation where a command has two inputs:
* `input1` has one possibility:
  * tags `A`, `B` and `C`
* `input2` has two possibilities:
  * `A`, `B`, `D`
  * `A`, `B`, `E`, `F`
```
the current `develop` approach would consider the scores for these two equal (both have two matching) - so it errors out.
My change makes it so it's also inversely proportional to the difference between the tags.

So, in this example, the `input2-1` would get score `[false, -2, 2]`, and `input2-2` - score `[false, -2, 3]` - making the first one a clear winner.

Reply:
Provide a concrete example in the codebase where not having this is a problem.
I'm striving to really *not have* to rely on more complex features in there and find alternative solutions.

Reply:
I _really_ should start double-checking these before asking for reviews. It looks like some unrelated change hid this issue too. So I'm closing this thread for now.